apps/system/test/unit/nfc_handover_manager_test.js

```diff
@@ -93,14 +86,280 @@ suite('Nfc Handover Manager Functions', function() {
 
   teardown(function() {
     settingsCore.stop();
-    spyDefaultAdapter.restore();
-    spyBluetoothPair.restore();
   });
 
-  var invokeBluetoothGetDefaultAdapter = function() {
-    var adapterRequest = spyDefaultAdapter.firstCall.returnValue;
-    adapterRequest.fireSuccess(MockBluetooth.defaultAdapter);
-  };
+  suite('Initialize', function() {
+    teardown(function() {
+      nfcHandoverManager.stop();
+    });
+
+    test('start', function() {
+      this.sinon.stub(window.navigator, 'mozSetMessageHandler');
+      nfcHandoverManager.start();
+
+      assert.isFalse(nfcHandoverManager.incomingFileTransferInProgress);
+      assert.isFalse(nfcHandoverManager.bluetoothStatusSaved);
+      assert.isFalse(nfcHandoverManager.bluetoothAutoEnabled);
+      assert.ok(window.navigator.mozSetMessageHandler
+        .calledWith('nfc-manager-send-file'));
+    });
+
+    test('SetMessageHandler', function() {
+      nfcHandoverManager.start();
+      this.sinon.stub(nfcHandoverManager, 'handleFileTransfer');
+      var fileRequest = this.sinon.stub();
+      MockNavigatormozSetMessageHandler.mTrigger(
+        'nfc-manager-send-file', fileRequest);
+
+      assert.ok(nfcHandoverManager.handleFileTransfer
+        .calledWith(fileRequest));
+    });
+  });
+
+  suite('Events', function() {
+    teardown(function() {
+      nfcHandoverManager.stop();
+    });
+
+    test('nfc-transfer-started event is handled', function() {
+      this.sinon.spy(nfcHandoverManager, '_transferStarted');
+      nfcHandoverManager.start();
+      window.dispatchEvent(new CustomEvent('nfc-transfer-started',
+        {detail: {}}));
+
+      assert.ok(nfcHandoverManager._transferStarted.called);
+    });
+
+    test('nfc-transfer-completed event is handled', function() {
+      this.sinon.spy(nfcHandoverManager, 'transferComplete');
+      nfcHandoverManager.start();
+      window.dispatchEvent(new CustomEvent('nfc-transfer-completed',
+        {detail: {}}));
+
+      assert.ok(nfcHandoverManager.transferComplete.called);
+    });
+
+    test('bluetooth-disabled event is handled', function() {
+      this.sinon.spy(nfcHandoverManager, '_clearBluetoothStatus');
+      nfcHandoverManager.start();
+      window.dispatchEvent(new CustomEvent('bluetooth-disabled'));
+
+      assert.ok(nfcHandoverManager._clearBluetoothStatus.called);
+    });
+
+    test('adapter is not retrieved', function() {
```

Comment:
I think it would make sense to check few more things in this scenario. Let's add a fake action to the actionQueue with a stubbed callback. The test should check if the callback wasn't called and actionQueue wasn't emptied.